### PR TITLE
MO data delivery 202307 - add test annual meantemp files

### PR DESCRIPTION
**Pull request by mo-stephenpackman**

### CODE REVIEW

**Comment by mo-stephenpackman** [comment]
Annual mean temperature data file and configuration JSON files have been added to this PR. CSV cubed build ran successfully from the codespace and the resulting output files were generated.

However, in the build process, the measure column values were converted to lower case so we ended up with "mean" and "trend". This did not seem to cause an issue with the fact that the measure labels in the config file had these as "Mean" and "Trend" and the correct metadata seemed to be added. However, for consistency, I will update the labels in our local config files to lower case.

The annual mean temperature data has been reformatted to include a Measure column with mean and trend values separated here rather than in the Geography column.

File: annual_meantemp.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        "Geography": {
+            "label": "Met Office Geography",
+            "from_existing": "http://purl.org/linked-data/sdmx/2009/dimension#refArea",
+            "type": "dimension"
+        },
+        "Measure": {
+            "type": "measures",
+            "values": [
+                {
+                    "label": "Mean",
```
Comment (canwaf) [inline annual_meantemp.json]: Should say Mean Temperature

File: annual_meantemp.json [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                    "description": "The annual mean temperature, averaged out for the reporting period"
```
Comment (canwaf) [inline annual_meantemp.json]: Add a from_existing: for our temperature measure

File: annual_meantemp.json [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+                },
+                {
+                    "label": "Trend",
```
Comment (canwaf) [inline annual_meantemp.json]: The trend is a temperature still

File: annual_meantemp.json [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+                    "description": "Climatological trend calculated using triangular window method for each geographical area"
```
Comment (canwaf) [inline annual_meantemp.json]: Add a from_existing related to temperature again

File: annual_meantemp.json [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+                }
+            ]
+        },
+        "Value": {
+            "type": "observations",
+            "unit": {
+                "label": "Celsius",
```
Comment (canwaf) [inline annual_meantemp.json]: Remove the label, it's implied when you use Celsius directly.

File: annual_meantemp.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+    "keywords": [
+        "climate",
+        "temperature"
+    ],
+    "columns": {
+        "Year": {
+            "from_template": "year"
+        },
+        "Geography": {
+            "label": "Met Office Geography",
```
Comment (canwaf) [inline annual_meantemp.json:20]: Do these directly map to the administrative geographies of England, Wales, Scotland, and Northern Ireland? If so, I suggest you use the ONS Geography codes.

E09000001, etc.

**Review by canwaf — CHANGES_REQUESTED** [review]

File: annual_meantemp.json (hunk 3 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json]: Shall I use Temperature Trend as the label instead?

File: annual_meantemp.json (hunk 6 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json:20]: We can map the Geography names to the ONS country codes in the CSV input files if that makes it easier for you to decode at your end.

I.e.

UK: K02000001
England: E92000001
Wales: W92000004
Scotland: S92000003
Northern Ireland: N92000002

If we do this, do we also need to update the Geography from_existing link?

Comment (mo-stephenpackman) [reply annual_meantemp.json:20]: Country codes now used instead of country name in Geography column.

File: annual_meantemp.json (hunk 5 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json]: Label removed for value unit.

File: annual_meantemp.json (hunk 4 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json]: From_existing tag added to temperature and trend labels.

File: annual_meantemp.json (hunk 2 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json]: From_existing tag added to temperature and trend labels.

File: annual_meantemp.json (hunk 3 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json]: Label updated.

File: annual_meantemp.json (hunk 1 above)
Comment (mo-stephenpackman) [reply annual_meantemp.json]: Label updated.

**Comment by mo-stephenpackman** [comment]
- Annual CSV input file now uses ONS country codes instead of geographical country name.
- Configuration file has been updated following comments: labels updated on Mean Temperature and Temperature Trend measures, from_existing tags added and label dropped on units.

**Pull request merged**